### DISCUSSION TITLE
Allow @tailwindcss/upgrade to codemod 4.0 alpha projects

### DISCUSSION THREAD

**Discussion by bai** [tailwindlabs/tailwindcss#15080] [General]
I've been trying alphas for past couple months and now that the beta is out, wanted to run `npx @tailwindcss/upgrade@next` to ensure I use new classnames (e.g. `ring-3` vs `ring`) everywhere. However, I'm getting the following error:

```
pnpx @tailwindcss/upgrade@next
≈ tailwindcss v4.0.0-beta.1

│ Tailwind CSS v4.0.0-beta.1 found. The migration tool can only be run on v3 projects. 
```

Thoughts on allowing codemod to run on 4.0 alpha codebase?

**Comment by philipp-spiess** [comment]
@bai Unfortunately this is not supported right now because some of the codemods we include would not be safe to run multiple times (e.g. the variant order change https://tailwindcss.com/docs/v4-beta#stacking-order-sensitive-variants). If you've already been running v4 alpha, there shouldn't really be any changes necessary to upgrade to the beta, though!
